### PR TITLE
refactor(debugger): remove attr usage

### PR DESCRIPTION
**Pull request by P403n1x87**
We refactor the debugger code to remove any usages of the attr package in favour of the built-in dataclasses library.

## Checklist

- [x] The PR description includes an overview of the change
- [x] The PR description articulates the motivation for the change
- [x] The change includes tests OR the PR description describes a testing strategy
- [x] The PR description notes risks associated with the change, if any
- [x] Newly-added code is easy to change
- [x] The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- [x] The change includes or references documentation updates if necessary
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Newly-added code is easy to change
- [x] Release note makes sense to a user of the library
- [x] If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/debugging/_probe/model.py [hunk 1]
```diff
@@ -79,68 +79,77 @@ def update(self, other: "Probe") -> None:
         if self.version == other.version:
             return
 
-        for attrib in (_.name for _ in self.__attrs_attrs__ if _.eq):
+        for attrib in (f.name for f in self.__dataclass_fields__.values() if f.compare):
```
Comment (gnufede) [inline ddtrace/debugging/_probe/model.py:82]: I think `dataclasses.fields(self)` would be equivalent to `self.__dataclass_fields__.values()`

File: ddtrace/debugging/_probe/model.py [hunk 2]
```diff
@@ -79,68 +80,77 @@ def update(self, other: "Probe") -> None:
...
+@dataclass
 class LineLocationMixin(ProbeLocationMixin):
-    source_file = attr.ib(type=Path, converter=_resolve_source_file, eq=False)  # type: ignore[misc]
-    line = attr.ib(type=int, eq=False)
+    source_file: str = field(compare=False)
+    line: int = field(compare=False)
+
+    def __post_init__(self):
+        super().__post_init__()
+        self.resolved_source_file = _resolve_source_file(self.source_file)
```
Comment (shatzi) [inline ddtrace/debugging/_probe/model.py:150]: can we add this logic into source_file init ? 
maybe we also like to keep the original value (aka, when we report errors or when _resolve_source_file fails)

File: ddtrace/debugging/_probe/model.py [hunk 3]
```diff
@@ -79,68 +80,77 @@ def update(self, other: "Probe") -> None:
...
 
     def location(self):
-        return (str(self.source_file) if self.source_file is not None else None, self.line)
+        return (str(self.resolved_source_file) if self.source_file is not None else None, self.line)
```
Comment (shatzi) [inline ddtrace/debugging/_probe/model.py:153]: if self.source_file or this need to check resolved_source_file instead?

**Review by shatzi — APPROVED** [review]
awesome stuff

File: ddtrace/debugging/_probe/model.py (hunk 2 above)
Comment (P403n1x87) [reply ddtrace/debugging/_probe/model.py:150]: unfortunately `dataclasses` don't have the ability of using a custom converter so we have to use an extra field here

File: ddtrace/debugging/_probe/model.py (hunk 3 above)
Comment (P403n1x87) [reply ddtrace/debugging/_probe/model.py:153]: I think we can probably remove the `if` now 🤔  as `self.source_file` should never be `None`

**Pull request merged**